engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/AvroSchemaDeterminer.scala

```diff
@@ -9,11 +9,13 @@ trait AvroSchemaDeterminer extends Serializable {
 
   /**
    * This method should provide schema that will be used in runtime. In some cases it can be other than schema used
-   * in typing. For example when you want to use schema of record that will be consumed in source and have a typing
-   * just as a addition that have little influence on processing. See `UsingRecordSchemaInRuntimeAvroSchemaDeterminer`
-   * for more info.
+   * in typing. For example:
+   * - I have #input of type schemaA
+   * - I want to to typed filtering on #input.field1
+   * - I want to pass #input to sink, with all (possibly unknown at deployment time) fields
    */
-  def determineSchemaInRuntime: Validated[SchemaDeterminerError, Option[Schema]]
+  def toRuntimeSchema(schemaUsedInTyping: Schema): Option[Schema] = Some(schemaUsedInTyping)
```

Comment:
Does it realy have sens?

Reply:
It is default behaviour. After changing it to `None` you will achieve behavior that you've called before this change `(SchemaDeterminingStrategy.FromRecord)` - you have tests for that so it looks like it is important :) BTW take a look at doc above

Reply:
Okey but, right now you run it by:
```
val schema = determiner.determineSchemaInRuntime.get
val runTimeSchema = determiner.determineSchemaInRuntime(schema)
```
I think bettter way is do it in trait,  I mean:
```
def toRuntimeSchema(schemaUsedInTyping: Schema): Option[Schema]  = Some(
determineSchemaInRuntime.getWithCatching
)
```
I someone want change it then he overrride it.

Reply:
It won't be good because take a look that we always fetch latest schema for typing. Cashing could introduce unexpected desynchronizations between those two schema. I think that determining runtime schema should take previously determined schema for typing